pulp_container/app/viewsets.py

```diff
@@ -587,6 +589,29 @@ class ContainerPushRepositoryViewSet(ReadOnlyRepositoryViewSet):
         ],
     }
 
+    def get_queryset(self):
```

Comment:
I had to add custom filtering for the repositories so that we could consider permissions on the namespace and the private field on the distribution. We will need to do something similar for the distributions viewset.

Reply:
+1 not necessarily part of this pr, but commenting now for the sake of capturing a TODO

Reply:
i filed an issue https://pulp.plan.io/issues/8206